lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/ApacheParquetValueDecoders.java

```diff
@@ -505,6 +507,82 @@ public void skip(int n)
         }
     }
 
+    public static final class DoubleApacheParquetValueDecoder
+            implements ValueDecoder<long[]>
+    {
+        private final ValuesReader delegate;
+
+        public DoubleApacheParquetValueDecoder(ValuesReader delegate)
+        {
+            this.delegate = requireNonNull(delegate, "delegate is null");
+        }
+
+        @Override
+        public void init(SimpleSliceInputStream input)
+        {
+            byte[] buffer = input.readBytes();
+            try {
+                int valueCount = buffer.length / Byte.SIZE;
+                delegate.initFromPage(valueCount, ByteBufferInputStream.wrap(ByteBuffer.wrap(buffer, 0, buffer.length)));
+            }
+            catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
```

Comment:
re-use existing `initialize` method from this class

Reply:
@raunaqmorarka Just wanted to bring up that existing initialize (https://github.com/trinodb/trino/blob/504fab7882e29454fd54f23b1a72fd1e72cbe25f/lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/ApacheParquetValueDecoders.java#L111) initializes valuesCount to 0 in the call to ByteStreamSplitValuesReader::initFromPage. But initFromPage in ByteStreamSplitValuesReader calculates and validates valuesCount (https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesReader.java#L70).
This is causing failures in my tests in https://github.com/trinodb/trino/pull/12809/files#diff-912cf32303d6b41c24cc3093108f5e1edf7e3c260dd3d5b147756af84bef031fR1906 and https://github.com/trinodb/trino/pull/12809/files#diff-912cf32303d6b41c24cc3093108f5e1edf7e3c260dd3d5b147756af84bef031fR1932. 

Anyhow I will re-write these tests to use pre-generated parquet file. If I still see problem in those tests, I will have to keep a custom init method.